controllers/binding_controller_test.go

```diff
@@ -0,0 +1,217 @@
+				}, nil)
+			})
+
+			It("sets the status condition to indicate a success in reconciling", func() {
+				Expect(client.Create(ctx, &binding)).To(Succeed())
+				EventuallyWithOffset(1, func() []topology.Condition {
+					_ = client.Get(
+						ctx,
+						types.NamespacedName{Name: binding.Name, Namespace: binding.Namespace},
+						&binding,
+					)
+
+					return binding.Status.Conditions
+				}, 10*time.Second, 1*time.Second).Should(ContainElement(MatchFields(IgnoreExtras, Fields{
+					"Type":   Equal(topology.ConditionType("Ready")),
+					"Reason": Equal("SuccessfulCreateOrUpdate"),
+					"Status": Equal(corev1.ConditionTrue),
+				})))
+			})
+		})
+	})
+
+	When("Deleting a binding", func() {
+		JustBeforeEach(func() {
+			fakeRabbitMQClient.DeclareBindingReturns(&http.Response{
+				Status:     "201 Created",
+				StatusCode: http.StatusCreated,
+			}, nil)
+			Expect(client.Create(ctx, &binding)).To(Succeed())
+			EventuallyWithOffset(1, func() []topology.Condition {
+				_ = client.Get(
+					ctx,
+					types.NamespacedName{Name: binding.Name, Namespace: binding.Namespace},
+					&binding,
+				)
+
+				return binding.Status.Conditions
+			}, 10*time.Second, 1*time.Second).Should(ContainElement(MatchFields(IgnoreExtras, Fields{
+				"Type":   Equal(topology.ConditionType("Ready")),
+				"Reason": Equal("SuccessfulCreateOrUpdate"),
+				"Status": Equal(corev1.ConditionTrue),
+			})))
+		})
+
+		When("the RabbitMQ Client returns a HTTP error response", func() {
+			BeforeEach(func() {
+				bindingName = "delete-binding-http-error"
+				fakeRabbitMQClient.DeleteBindingReturns(&http.Response{
+					Status:     "502 Bad Gateway",
+					StatusCode: http.StatusBadGateway,
+					Body:       ioutil.NopCloser(bytes.NewBufferString("Hello World")),
+				}, nil)
+			})
+
+			It("raises an event to indicate a failure to delete", func() {
+				Expect(client.Delete(ctx, &binding)).To(Succeed())
+				Consistently(func() bool {
+					err := client.Get(ctx, types.NamespacedName{Name: binding.Name, Namespace: binding.Namespace}, &topology.Binding{})
+					return apierrors.IsNotFound(err)
+				}, 5).Should(BeFalse())
+				Expect(observedEvents()).To(ContainElement("Warning FailedDelete failed to delete binding"))
+			})
+		})
+
+		When("the RabbitMQ Client returns a Go error response", func() {
+			BeforeEach(func() {
+				bindingName = "delete-binding-go-error"
+				fakeRabbitMQClient.DeleteBindingReturns(nil, errors.New("some error"))
+			})
+
+			It("raises an event to indicate a failure to delete", func() {
+				Expect(client.Delete(ctx, &binding)).To(Succeed())
+				Consistently(func() bool {
+					err := client.Get(ctx, types.NamespacedName{Name: binding.Name, Namespace: binding.Namespace}, &topology.Binding{})
+					return apierrors.IsNotFound(err)
+				}, 5).Should(BeFalse())
+				Expect(observedEvents()).To(ContainElement("Warning FailedDelete failed to delete binding"))
+			})
+		})
+
+		When("the RabbitMQ Client fails to generate", func() {
```

Comment:
I had to read the function `prepareNoSuchClusterError()` to understand the intention behind "when the RabbitMQ Client fails to generate". Perhaps we can improve the wording of this `when`?
Perhaps be super specific, like "when the RabbitMQ cluster is nil"?